### PR TITLE
Add test cases for required-optional.json , body-time.json , body-array.json

### PR DESCRIPTION
**Pull request by sacheu**
Fix Autorest Swift to support body-time.json.
Add new cases for required-optional.json , body-time.json , body-array.json

### CODE REVIEW

**Review by tjprescott — CHANGES_REQUESTED** [review]
One small nit.

File: AutorestSwiftTest/AutoRestRequiredOptionaltTest.swift [hunk 1]
```diff
@@ -0,0 +1,282 @@
+// --------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (sacheu) [reply AutorestSwiftTest/AutoRestRequiredOptionaltTest.swift]: Thanks. Push a new iteration with the fix

**Review by tjprescott — APPROVED** [review]

**Comment by tjprescott** [comment]
One test is failing, which I commented out. I will address it on the `dev/AzureCore` branch.

**Comment by sacheu** [comment]
Fix AB#2321068

**Pull request merged**